docs-ui/components/code.tsx

```diff
@@ -0,0 +1,142 @@
+import {withTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import copy from 'copy-text-to-clipboard';
+import Prism from 'prismjs';
+import 'prismjs/themes/prism.css';
+import {Component, createRef, RefObject, useEffect} from 'react';
+
+import {IconCode} from 'app/icons';
+import space from 'app/styles/space';
+import {Theme} from 'app/utils/theme';
+
+type Props = {
+  theme: Theme;
+  /**
+   * Main code content gets passed as the children prop
+   */
+  children: string;
+  /**
+   * Auto-generated class name for <pre> and <code> element,
+   * with a 'language-' prefix, e.g. language-css
+   */
+  className?: string;
+  /**
+   *  Meta props from the markdown syntax,
+   *  for example, in
+   *
+   * ```jsx label=hello
+   * [some code]
+   * ```
+   *
+   * the label prop is set to 'hello'
+   */
+  label?: string;
+};
+
+const Code = ({theme, children, className, label}: Props) => {
+  const codeRef: RefObject<HTMLElement> = createRef();
+  const copyButtonRef: RefObject<HTMLButtonElement> = createRef();
+
+  function copyCode() {
+    const copyButton = copyButtonRef?.current;
+
+    if (copyButton) {
```

Comment:
You could inverse this condition and make it a [guard statement](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html), which would reduce a level of indentation :)

```tsx
function copyCode() {
	const copyButton = copyButtonRef?.current;

	if (!copyButton) {
		return
	}

	...
}
```

Reply:
Ahh that does look nicer!